src/other/codemirror/CodeMirror.tsx

```diff
@@ -35,54 +36,56 @@ export interface CodeMirrorProps {
 }
 
 export interface CodeMirrorState {
-    content: string;
-    checks?: ConsistencyCheck[];
+    entity: { content: string; checks?: ConsistencyCheck[] };
     isEditing: boolean;
 }
 
-class CodeMirror extends React.Component<CodeMirrorProps, CodeMirrorState> {
+const CodeMirrorClass = React.Component<CodeMirrorProps, CodeMirrorState>;
+
+class CodeMirror extends StatefulEntityMixin(CodeMirrorClass) {
     constructor(props: CodeMirrorProps) {
         super(props);
         this.state = {
-            content: props.content || "",
-            checks: props.checks,
+            entity: { content: props.content || "", checks: props.checks },
             isEditing: false,
         };
         this.handleContentChange = this.handleContentChange.bind(this);
     }
 
     UNSAFE_componentWillReceiveProps(nextProps: CodeMirrorProps) {
-        const { content, checks } = this.props;
-        const update = {};
+        const { content, checks } = this.state.entity || {};
+        const entityUpdate = { ...this.state.entity };
+
         if (nextProps.content !== content) {
-            Object.assign(update, { content: nextProps.content || "" });
+            entityUpdate.content = nextProps.content || "";
         }
         if (nextProps.checks !== checks) {
-            Object.assign(update, { checks: nextProps.checks });
+            entityUpdate.checks = nextProps.checks;
         }
-        this.setState(update);
+
+        this.setState({ entity: entityUpdate });
     }
 
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
     shouldComponentUpdate(
         nextProps: Readonly<CodeMirrorProps>,
         nextState: Readonly<CodeMirrorState>,
         nextContext: never,
     ): boolean {
-        const { content, checks } = this.state;
-        const { content: nextContent, checks: nextChecks } = nextState;
-        return content !== nextContent || checks !== nextChecks;
+        const { entity } = this.state;
+        const { entity: nextEntity } = nextState;
+
+        return entity.content !== nextEntity.content || entity.checks !== nextEntity.checks;
     }
 
     handleContentChange(newContent: string) {
         const { isEditing } = this.state;
         const { updateContent } = this.props;
         if (isEditing && updateContent) updateContent(newContent);
-        this.setState({ content: newContent });
+        this.setState({ entity: { content: newContent } });
```

Comment:
Here the checks get overwritten and do not chow up. But if I add `...prevState.checks`, or `state.entity.checks` -- the whole codemirror rerenders moving cursor at the beginning. 
Don't want to get stuck on this

Reply:
{entity.content: newContent}

Reply:
Works!